### DISCUSSION TITLE
platformAutomerge does not set auto-complete on Azure Devops

### DISCUSSION THREAD

**Discussion by FrankHoogmans** [renovatebot/renovate#12881] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

Latest (npx renovate)

### Please select which platform you are using if self-hosting.

Azure DevOps (dev.azure.com)

### If you're self-hosting Renovate, tell us what version of the platform you run.

Latest (npx renovate)

### Describe the bug

I can not get the PR's to be set to auto-complete with any configuration.
This is a very minimal repro setup: https://dev.azure.com/hoogmans/RenovateTest which shows the issue.
The following PR will be created, but no auto-complete is set: https://dev.azure.com/hoogmans/RenovateTest/_git/RenovateTest/pullrequest/16

It has been broken since the following PR (https://github.com/renovatebot/renovate/pull/12069) has deprecated the `azureAutoComplete` flag, in favor of the new `platformAutomerge`. 
Before, we used `azureAutoComplete` with good success, auto-complete was set on the PR, but just on the default merge strategy.

Note, auto-merging is actually impossible because of policies, but before this change, the PR was actually set to auto-complete, which was our desired behavior.

Is my configuration wrong, was the auto-complete for Azure Devops deprecated or is there a bug somewhere?
Config:
``` 
module.exports = {
    platform: 'azure',
    endpoint: 'https://dev.azure.com/hoogmans/',
    token: process.env.TOKEN,
    hostRules: [],
    azureWorkItemId: 50,
    automerge: true,
    automergeType: 'pr',
    automergeStrategy: 'squash',
    platformAutomerge: true,
    repositories: ['RenovateTest/RenovateTest'],
  };
``` 

For context, the following PR and issue might be related...
https://github.com/renovatebot/renovate/pull/9702
https://github.com/renovatebot/renovate/issues/9685

### Relevant debug logs

<details><summary>Logs</summary>
```
DEBUG: Checking if we can automerge branch (repository=RenovateTest/RenovateTest, branch=renovate/repodb-1.x)
DEBUG: mergeStatus=no automerge (repository=RenovateTest/RenovateTest, branch=renovate/repodb-1.x)
```

```
DEBUG: PR is configured for automerge (repository=RenovateTest/RenovateTest, branch=renovate/repodb-1.x)
DEBUG: PR is not ready for merge (repository=RenovateTest/RenovateTest, branch=renovate/repodb-1.x)
       "canMergeReason": undefined

```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

**Comment by rarkins** [comment]
Did you read the 28.0.0 release notes regarding this feature? https://github.com/renovatebot/renovate/releases/tag/28.0.0

**Comment by FrankHoogmans** [comment]
> Did you read the 28.0.0 release notes regarding this feature? https://github.com/renovatebot/renovate/releases/tag/28.0.0

Yes. I have tried several combinations of `azureAutoComplete` and `automerge` (and related automerge options), but I can't get the PR set to auto-complete.
I have updated and tested the minimal repro to also include the `azureAutoComplete` flag again, but to no avail.

**Comment by rarkins** [comment]
Note: Autocomplete is set only at PR creation time. Can you do something to trigger a new PR while logging at debug log level, and then copy paste the logs for that particular branch here?

**Comment by FrankHoogmans** [comment]
> Note: Autocomplete is set only at PR creation time. Can you do something to trigger a new PR while logging at debug log level, and then copy paste the logs for that particular branch here?

I'm going to try that :)

**Comment by FrankHoogmans** [comment, marked as answer]
> Note: Autocomplete is set only at PR creation time. Can you do something to trigger a new PR while logging at debug log level, and then copy paste the logs for that particular branch here?

@rarkins I have tried your suggestion and it works 👍

So, anyone needing this: 
Enabling both `automerge` and `azureAutocomplete` resolved my issue.

**Reply by rarkins** [reply]
Relevant release notes:
![image](https://user-images.githubusercontent.com/6311784/143869609-73a4ec92-108c-416f-9479-4ae6678ed90e.png)
